text/0000-path-clarity.md

```diff
@@ -0,0 +1,621 @@
+
+Let's revisit one of the motivating examples. Today, you might write:
+
+```rust
+extern crate futures;
+fn my_poll() -> futures::Poll { ... }
+```
+
+and then be confused when the following doesn't work:
+
+```rust
+extern crate futures;
+mod submodule {
+    fn my_poll() -> futures::Poll { ... }
+}
+```
+
+because you've been led to think that `extern crate` brings the name into scope
+everywhere.
+
+After this RFC, you would no longer write `extern crate futures`. You might try to write just:
+
+```rust
+fn my_poll() -> futures::Poll { ... }
+```
+
+but the compiler would produce an error, saying that there's no `futures` in
+scope; maybe you meant the external dependency, which you can bring into scope
+by writing `use futures;`? So you do that:
+
+```rust
+use futures;
+fn my_poll() -> futures::Poll { ... }
+```
+
+and now, when you refactor, you're much more likely to understand that the `use`
+should come along for the ride:
+
+```rust
+mod submodule {
+    use futures;
+    fn my_poll() -> futures::Poll { ... }
+}
+```
+
+Together with the fact that you use `crate::` in `use` declarations, this
+strongly reinforces the idea that:
+
+- `use` brings items into scope, based on paths that start by identifying the crate
+- an item needs to be in scope before you can refer to it
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+First, a bit of terminology: a *fully qualified path* is a path starting with
+`::`, which *all* paths in `use` do implicitly.
+
+The actual changes in this RFC are fairly small tweaks to the current module
+system; most of the complexity comes from the migration plans.
+
+The proposed migration plan is minimally disruptive; **it does not require an
+epoch**.
+
+## Basic changes
+
+- You can write `mod bar;` statements even when not in a `mod.rs` or equivalent;
+  in this case, the submodules must appear within a subdirectory with
+  the same name as the current module. Thus, `foo.rs` can contain `mod bar;` if
+  there is also a `foo/bar.rs`.
+  - It is not permitted to have both `foo.rs` and `foo/mod.rs` at the same point
+    in the file system.
+  - The use of `mod.rs` continues to be allowed without any deprecation.
+
+- We introduce `crate` as a new visibility specifier, shorthand for `pub(crate)`
+  visibility.
+
+- We introduce a lint against use of bare `pub` for items which are not
+  reachable via some fully-`pub` path. That is, bare `pub` should truly mean
+  *public*, and `crate` should be used for crate-level visibility.
+
+- We introduce `crate` as a new path component which designates the root of the
+  current crate, and deprecate fully qualified paths that do not begin with one
+  of: an external crate name, `crate`, `super`, or `self`.
+
+- Cargo will provide a new `alias` key for dependencies, so that e.g. users who
+  want to use the `rand` crate but call it `random` instead can now write `random
+  = { version = "0.3", crate = "rand" }`.
```

Comment:
The paragraph starts out mentioning an alias key, but no such key comes up later. Looks like an incomplete update.